src/Modules/Manage/Dnn.PersonaBar.Roles/Components/RolesController.cs

```diff
@@ -64,6 +72,50 @@ public IEnumerable<RoleInfo> GetRoles(PortalSettings portalSettings, int groupId
             return roleInfos.Skip(startIndex).Take(pageSize);
         }
 
+        /// <summary>
+        /// Gets a list (not paginated) of Roles given a comma separated list of Roles' names.
+        /// </summary>
+        /// <param name="portalSettings"></param>
+        /// <param name="groupId"></param>
+        /// <param name="commaSeparatedRoles"></param>
+        /// <param name="notFound">List of "not found roles"</param>
+        /// <returns>List of found Roles</returns>
+        public List<RoleInfo> GetRolesByNames(PortalSettings portalSettings, int groupId, string commaSeparatedRoles, out List<string> notFound)
+        {
+            var isAdmin = IsAdmin(portalSettings);
+            HashSet<string> rolesList = new HashSet<string>();
+            if (!string.IsNullOrWhiteSpace(commaSeparatedRoles))
+            {
+                commaSeparatedRoles.Split(',').ToList().ForEach((item) => rolesList.Add(item.Trim()));
+            }
+
+            List<RoleInfo> foundRoles = null;
+            HashSet<string> foundNames = new HashSet<string>();
+            if (rolesList.Count() > 0)
+            {
+                var allRoles = (groupId < Null.NullInteger
+                ? RoleController.Instance.GetRoles(portalSettings.PortalId)
+                : RoleController.Instance.GetRoles(portalSettings.PortalId, r => r.RoleGroupID == groupId));
+                
+                foundRoles = allRoles.Where(r => 
+                {
+                    bool adminCheck = isAdmin || r.RoleID != portalSettings.AdministratorRoleId;
```

Comment:
Mm, I won't question this logic in this particular but I see it occurs in a number of methods in this class. Maybe we could extract it to a shared function?